### PR TITLE
Add Artforum.js

### PR DESCRIPTION
**Pull request by owcz**
Translator for art news website http://www.artforum.com. Covers major pages and a multi for search. Note that some pages are undated and others have no author, hence their absence in the test results.

### CODE REVIEW

**Comment by owcz** [comment]
## ready for review
* removed [Framework](https://www.zotero.org/support/dev/translators/framework) per #1277
* now expands on [EM](https://github.com/zotero/translators/blob/master/Embedded%20Metadata.js)
* multi support for homepage, many section pages, search
* open to ideas on how to pull the magazine's volume/issue numbers from their subpages

File: Artforum.js [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+		for (let author of authorMetadata) {
+			item.creators.push(ZU.cleanAuthor(author.text, "author"));
+		}
+		if (url.includes('/print/')) {
+			item.itemType = "magazineArticle";
+			item.ISSN = "0004-3532";
+			var issueDate = doc.querySelector('h3.print-article__issue-title');
+			if (issueDate) {
+				item.date = issueDate.textContent.trim().replace('PRINT ','');
+				/* are you feeling adventurous? could follow link to issue's page and scrape the volume/issue numbers...
```
Comment (adam3smith) [inline Artforum.js]: If you want to do this, I think the way to go is to reconstruct from the issue description: every year has ten issues from September to May, then a Summer issue, so getting the issue is easy. For the volume, you'd have to do something like 
take the year, subtract 1962, except for September, October, November December, where you subtract 1961. Making the extra call seems tedious, though also doable.

File: Artforum.js [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+		case "multiple":
+			Zotero.selectItems(getSearchResults(doc, false), function (items) {
+				if (!items) {
+					return true;
+				}
+				var articles = [];
+				for (var i in items) {
+					articles.push(i);
+				}
+				Z.debug(articles);
```
Comment (adam3smith) [inline Artforum.js]: remove

File: Artforum.js [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+				ZU.processDocuments(articles, scrape);
+			});
+			break;
+		case "blogPost":
```
Comment (adam3smith) [inline Artforum.js]: use default: here to also capture magazine articles once you update above

File: Artforum.js [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	if (/-\d{5,}$/.test(url)) {
```
Comment (adam3smith) [inline Artforum.js]: You should be able to detect `magazineArticle` by checking for the "Print" thing

File: Artforum.js [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+	});
+}
+
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	// 1st for search page, 2nd for issue ToC, 3rd/4th/5th for sections, 5th+ for homepage
+	var rows = doc.querySelectorAll('h1.results-list__h1, .toc-article__title, .news-list h1, .reviews-list h1, .article-list h1, p.hp-singlefeature-author__writer, h3.hp-news__title, h3.hp-twocolumn__title a, h3.hp-artguide__title, p.hp-bloglist__teaser a');
+	var links = doc.querySelectorAll('h1.results-list__h1 a, .toc-article__title a, .news-list h1 a, .reviews-list h1 a, .article-list h1 a, h3.hp-singlefeature-author__title > a, a.hp-news__article, h3.hp-twocolumn__title a, .hp-artguide div.image-container > a, p.hp-bloglist__teaser a');
```
Comment (owcz) [inline Artforum.js]: is this selector list too expensive? if preferable, i could break the list into subsets and first check against the page URL?

only downside is that this site has many eligible multi pages and i don't want to end up missing one

File: Artforum.js (hunk 1 above)
Comment (owcz) [reply Artforum.js]: i thought about the former but seemed like more of a hack and prone to potential error if the publication cycle was altered in a year

looking for a model: any recommendations for translators that call through to another page?

Comment (adam3smith) [reply Artforum.js]: This isn't a perfect example but basically what you'd do is along the lines of https://github.com/zotero/translators/blob/master/Stanford%20Encyclopedia%20of%20Philosophy.js#L71 
You scrape all the info you need from the current page, then do a `ZU.doGet` or (more likely here) `ZU.processDocuments` for the new page add the missing information to the item and (and this is important) finalize the item within that routine, i.e. `item.complete()` would be within the `ZU.processDocuments`.

Comment (owcz) [reply Artforum.js]: I couldn't perform querySelector/xpath on the doc so had to go with regex—not sure if that's slower

**Comment by owcz** [comment]
@adam3smith this should be ready for review (as are [the rest of these](https://github.com/zotero/translators/pulls?utf8=%E2%9C%93&q=is%3Aopen+is%3Apr+author%3Aowcz+NOT+ebscohost+))

**Pull request merged**